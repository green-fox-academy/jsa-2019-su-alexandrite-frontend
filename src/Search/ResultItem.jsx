import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Text,
  Image,
} from 'react-native';
import { useNavigation } from 'react-navigation-hooks';
import { TouchableHighlight } from 'react-native-gesture-handler';
import styles from './styles';

import tradeIcon from '../../assets/icons/search-page/trade.png';
import detailIcon from '../../assets/icons/search-page/detail.png';

const ResultItem = ({ symbol, exchange }) => {
  const { push } = useNavigation();
  return (
    <View style={styles.resultList}>
      <View>
        <Text style={styles.stockName}>{symbol}</Text>
        <Text style={styles.stockExchange}>{exchange}</Text>
      </View>
      <View style={styles.resultButton}>
        <TouchableHighlight>
          <Image source={tradeIcon} />
        </TouchableHighlight>
        <TouchableHighlight
          style={styles.detailButton}
          onPress={() => push('StockDetails', { symbol })}
          activeOpacity={0.5}
          underlayColor="#0000"
        >
          <Image source={detailIcon} />
        </TouchableHighlight>
      </View>
    </View>
  );
};

ResultItem.propTypes = {
  symbol: PropTypes.string.isRequired,
  exchange: PropTypes.string.isRequired,
};

export default ResultItem;
