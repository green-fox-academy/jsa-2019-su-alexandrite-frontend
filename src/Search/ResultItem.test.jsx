import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { useNavigation } from 'react-navigation-hooks';
import ResultItem from './ResultItem';

jest.mock('react-redux');
jest.mock('../redux/search/actionCreator');
jest.mock('react-navigation-hooks');

describe('<ResultItem />', () => {
  beforeEach(() => {
    useNavigation.mockReturnValue({
      push: jest.fn(),
    });
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<ResultItem symbol="AAPL" exchange="test" />);
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const tree = renderer.create(
      <ResultItem symbol="AAPL" exchange="test" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should call navigation.push and pass \'StockDetails\' and {symbol}', () => {
    const mockPush = jest.fn();
    useNavigation.mockReturnValueOnce({
      push: mockPush,
    });
    const wrapper = shallow(
      <ResultItem symbol="AAPL" exchange="test" />,
    );
    wrapper.find({ testID: 'details-button' }).first().simulate('press');
    expect(mockPush).toHaveBeenCalledWith('StockDetails', { symbol: 'AAPL' });
  });
});
