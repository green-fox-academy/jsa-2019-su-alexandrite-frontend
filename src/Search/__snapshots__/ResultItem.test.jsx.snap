// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultItem /> Should render and match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#eeeeee",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 17,
      "paddingVertical": 11,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      AAPL
    </Text>
    <Text
      style={
        Object {
          "color": "#999999",
          "fontSize": 10,
        }
      }
    >
      test
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": -10,
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={Object {}}
      >
        <Image
          source={1}
          style={null}
        />
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "marginLeft": 26,
            "padding": 10,
          }
        }
        testID="details-button"
      >
        <Image
          source={1}
          style={null}
        />
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
